src/app/ffbe/unit-awakening-materials-display/unit-awakening-materials-display.component.ts

```diff
@@ -0,0 +1,88 @@
+import {Component, Input, OnChanges, OnInit} from '@angular/core';
+import {Unite} from '../model/unite.model';
+import {Objet} from '../model/objet.model';
+import {UniteMateriauEveil} from '../model/unite-materiau-eveil.model';
+import {FfchClientService} from '../services/ffch-client.service';
+import {isNullOrUndefined} from 'util';
+import {UniteMateriauxEveil} from '../model/unite-materiaux-eveil.model';
+import {FfbeUtils} from '../utils/ffbe-utils';
+
+@Component({
+  selector: 'app-unit-awakening-materials-display',
+  templateUrl: 'unit-awakening-materials-display.component.html',
+  styleUrls: ['unit-awakening-materials-display.component.css']
+})
+export class UnitAwakeningMaterialsDisplayComponent implements OnInit, OnChanges {
+
+  @Input() unite: Unite;
+  public materiauxEveilErrors: Array<string> = [];
+  public materiauxEveilFromFfch: Array<UniteMateriauEveil> = [];
+
+  constructor(private ffchClientService: FfchClientService) {
+  }
+
+  ngOnInit() {
+    this.getObjets();
```

Comment:
À mettre dans le ngOnChanges sinon il y a un risque que this.unite.materiauxEveil.materiau soit null lorsqu'on insère une seconde unité (pas de re-trigger de ngOnInit tant que le composant est affiché dans la vue).